src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import { InputGroup, Button } from '../../index';
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+
+export const getActiveItems = items =>
+  items
+    .filter(
+      ({ disabled, type }) => !disabled && !['header', 'divider'].includes(type)
+    )
+    .map(({ text }) => text);
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
+          if (typeof inputValue === 'string') {
+            onInputUpdate(inputValue);
+            this.setState({ inputValue });
+          }
+        }}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          getLabelProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          getRootProps,
+          selectedItem,
+          selectItem
+        }) => (
+          <div>
+            {labelText && <label {...getLabelProps()}>{labelText}</label>}
+            <InputGroup>
+              <AutoCompleteInput
+                onKeyPress={e => {
```

Comment:
Let's try to keep this method lean.
Can you move this method into `this.handleKeyPress`?

Reply:
Same thing here with all the closure scope we need. It would be nice to isolate but I don't see how without forward a bunch of variables which I'm not a fan of.

Reply:
Yeah I'm not a fan of binding in this context. I think it's easier to see what's happening inline even though it's long.